### PR TITLE
Fixing #786

### PR DESCRIPTION
**Pull request by cube2222**

### CODE REVIEW

**Comment by dpiddy** [comment]
How about adding http2Server.readFrame which calls framer.readFrame, logs the error if it's not EOF, then returns it? The two current calls to framer.readFrame can be replaced with that which will collapse this a bit.

**Comment by cube2222** [comment]
After writing it down it seems to me that this would be inconsistent, as the rest of the logging in this file is also done in such top-level handlers.

**Comment by cube2222** [comment]
I signed it!

File: transport/http2_server.go [hunk 1]
```diff
@@ -265,6 +269,10 @@ func (t *http2Server) HandleStreams(handle func(*Stream)) {
 				t.controlBuf.put(&resetStream{se.StreamID, se.Code})
 				continue
 			}
+			if err == io.EOF || err == io.ErrUnexpectedEOF {
+				t.Close()
+				return
+			}
```
Comment (iamqizhao) [inline transport/http2_server.go]: The above block is not need because t.Close is called in line#277.

File: transport/http2_server.go [hunk 2]
```diff
@@ -239,6 +239,10 @@ func (t *http2Server) HandleStreams(handle func(*Stream)) {
 	}
 
 	frame, err := t.framer.readFrame()
+	if err == io.EOF || err == io.ErrUnexpectedEOF {
```
Comment (iamqizhao) [inline transport/http2_server.go]: this should be as same as line#262-277.

**Comment by cube2222** [comment]
In both cases it's appropriate, as the problem with this code is that it shouldn't log the error when it's EOF or unexpected EOF, because those are expected and will happen on proper requests. That's why we return early, before it gets to logging.

**Pull request merged**